core/azure-core/src/main/java/com/azure/core/polling/Poller.java

```diff
@@ -0,0 +1,281 @@
+     * @param cancelOperation cancel operation
+     */
+    public Poller(PollerOptions pollerOptions, Function<PollResponse<T>, Mono<PollResponse<T>>> pollOperation, Consumer<Poller> cancelOperation) {
+        this(pollerOptions, pollOperation);
+        this.cancelOperation = cancelOperation;
+    }
+
+    /**
+     * This will call cancelOperation function if provided. Once cancelOperation is triggered successfully, it can not be called again.
+     * This is to avoid unintentional calls to cancelOperation.
+     * It will not call cancelOperation if operation status is not started/Cancelled/Failed/successfully completed.
+     *
+     * @throws UnsupportedOperationException when cancel operation is not provided.
+     * @see com.azure.core.polling.PollResponse.OperationStatus
+     */
+    public void cancelOperation() throws UnsupportedOperationException {
+        if (cancelOperation == null) {
+            throw new UnsupportedOperationException("Cancel operation is not supported on this service/resource.");
+        }
+
+        // We can not cancel an operation if it was never started
+        // or it is in its terminal state.
+        // Check cancelInitiated: to protect against multiple time call to cancel operation
+        if (cancelInitiated || (pollResponse != null && pollResponse.getStatus() != PollResponse.OperationStatus.IN_PROGRESS)) {
+            return;
+        }
+        //Time to call cancel
+        cancelOperation.accept(this);
+        cancelInitiated = true;
+    }
+
+    /**
+     * Enable user to subscribe and receive all the responses.
+     * The user will start receiving PollResponse When client subscribe to this Flux.
+     * This Flux will not trigger a call to PollOperation.
+     * The poller still have its own default polling in action unless, user has  turned off
+     * auto polling.
+     *
+     * @return poll response as Flux that can be subscribed.
+     */
+    public Flux<PollResponse<T>> getObserver() {
+        return fluxHandle;
+    }
+
+    /**
+     * Calls poll operation once in sync if Pool operation is not completed.
+     *
+     * @return a Mono of {@link PollResponse}
+     */
+    public Mono<PollResponse<T>> poll() {
+
+        if (!isTerminalState()) {
+            updatePollOperationAsync();
+        }
+        return Mono.just(pollResponse);
+    }
+
+    /**
+     * This will block till poll operation is complete
+     *
+     * @return returns poll response
+     */
+    public PollResponse<T> block() {
+        return fluxHandle.blockLast();
+    }
+
+    /*
+     * Calls poll operation function and update pollResponse.
+     */
+    private void updatePollOperationAsync() {
+        pollOperation.apply(pollResponse).subscribe(pResponse -> {
+            processAsynchPollResponse(pResponse);
+        });
+    }
+
+    private void processAsynchPollResponse(PollResponse response) {
+        this.pollResponse = response;
+    }
```

Comment:
Get rid of this method, just do the one line operation directly on line 187 above.

Reply:
done